AR6 RF Update

kdorheim opened the pull request:
This is the PR it the work that updates Hector's RF to be consistent with the AR6 RF values [see here for full comparison](https://rpubs.com/kdorheim/838744) 

For the most part, I am satisfied with most of the updates, however, I do have some concerns about CO2, N2O, H2O strat, and aerosol-cloud interactions. I would really appreciate some feedback on forcing from aerosol-cloud interactions. 

Overview 
* RF CO2 ❓
* RF N2O ❓
* RF CH4 ✅
* RF O3 Trop 🆗 (no science update on Hector end) 
* RF H2O strat ❓ looks okay historical not great in the future periods
* RF Faci 👎
* RF Ari 🆗 (IPCC category of BC + OC + SO2 + NH3 RF)
* RF Fvol ✅
* RF F halocarbons 🆗
* RF Falbedo ✅
* RF Fmisc ✅

A comparison of the IPCC AR6 total RF driven Hector with emission and concentration driven Hector. 

![image](https://user-images.githubusercontent.com/27299759/142964514-bcc16c9d-d464-464e-b723-12df1ae8d362.png)


The difference in total RF ranges between 0 and 0.2 W m-2 which corresponds to a temperature one-tenth of a degree. Is am struggling to judge if this is okay or not... 

![image](https://user-images.githubusercontent.com/27299759/142964677-e19a2375-15c2-41ca-a833-8c24af728e03.png)


Changes 
* N2O & CH4 natural emissions 
* natural SO2 emissions removed 
* indirect and direct SO2 RF removed 
*  RF ACI & RF NH3 added
* Added miscellaneous RF category 
* switched to ssp scenario (note only added a single scenario to start)

Potential Issues 
* IPCC AR6 aerosols used different emission input files, however, the RF Ari looks good which leads me to believe that the historical emissions are not responsible for causing the difference in RF ACI 

To Do 
* Determine if the updates to forcing components are good or not ** this is the priority and needs inputs from others
* Add the other ssp ini files 
* update vignettes & testing files to use the ssps

ssmithClimate commented:
Took a quick look through and my main comment is that this is really great work! Nothing jumped out at me immediately as a huge issue. I see you had some specific comments, I will look through in more detail as needed.

kdorheim commented:
@ssmithClimate thanks! Could you please take a look at the RF ACI? and the other aerosol radiative forcings?

bpbond commented on inst/include/h_util.hpp:
```diff
@@ -28,6 +28,6 @@
 #define MODEL_VERSION "2.5.0"
 
-#define OUTPUT_DIRECTORY "output/"
+#define OUTPUT_DIRECTORY "/Users/dorh012/Desktop/"
```
Um we probably don't want this :)

bpbond replied:
Is this when working in R? Output is definitely disabled in that case. Or a C++ side oddity?

kdorheim replied:
No it does not work on the R side of things.

bpbond replied:
No, it shouldn't; the csvoutputstream visitor is disabled when running through R.

kdorheim replied:
Do we expect any output to be written when running from Xcode or executing from the command line?

bpbond replied:
Yes

bpbond commented on inst/include/h_util.hpp:
```diff
@@ -28,6 +28,6 @@
  */
 #define MODEL_VERSION "2.5.0"
```
Why don't you update this to 3.0 while you're at it?

kdorheim replied:
Sounds good, it has been updated & I added the v3.0 git tag

bpbond replied:
That feels momentous 👏

bpbond commented on src/ocean_csys.cpp:
```diff
@@ -158,7 +158,7 @@ void oceancsys::ocean_csys_run( unitval tbox, unitval carbon )
 
 	// Check that all is OK with input data
-	H_ASSERT( Tk > 265 && Tk < 308, "bad Tk value" ); // Kelvin
+	//H_ASSERT( Tk > 265 && Tk < 308, "bad Tk value" ); // Kelvin
```
Why commenting this out?

kdorheim replied:
I think this was a left over from when I was debugging something.

bpbond reviewed (changes requested):
I agree with Steve's comment above—great work, Kalyn. Kind of a slog and unglamorous but crucial for Hector v3 🥂

kdorheim commented:
Hi Ben,

Ugh why can’t I find this comment on github? No it does not work on the R side of things.







From: Ben Bond-Lamberty ***@***.***>
Date: Tuesday, January 4, 2022 at 9:23 AM
To: JGCRI/hector ***@***.***>
Cc: Dorheim, Kalyn R ***@***.***>, Author ***@***.***>
Subject: Re: [JGCRI/hector] AR6 RF Update (PR #536)
Check twice before you click! This email originated from outside PNNL.


@bpbond commented on this pull request.

________________________________

In inst/include/h_util.hpp<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FJGCRI%2Fhector%2Fpull%2F536%23discussion_r778118692&data=04%7C01%7Ckalyn.dorheim%40pnnl.gov%7C1283c24ad9cd4e88417a08d9cf8dd10c%7Cd6faa5f90ae240338c0130048a38deeb%7C0%7C0%7C637769030268474803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=LzJFQHa7erd8a%2FNGOKVvwRg7uD3AqR7JN3vDh%2BqCWYQ%3D&reserved=0>:

> @@ -28,6 +28,6 @@

  */

 #define MODEL_VERSION "2.5.0"



-#define OUTPUT_DIRECTORY "output/"

+#define OUTPUT_DIRECTORY "/Users/dorh012/Desktop/"

Is this when working in R? Output is definitely disabled in that case. Or a C++ side oddity?

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FJGCRI%2Fhector%2Fpull%2F536%23discussion_r778118692&data=04%7C01%7Ckalyn.dorheim%40pnnl.gov%7C1283c24ad9cd4e88417a08d9cf8dd10c%7Cd6faa5f90ae240338c0130048a38deeb%7C0%7C0%7C637769030268474803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=LzJFQHa7erd8a%2FNGOKVvwRg7uD3AqR7JN3vDh%2BqCWYQ%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGQI7LZJGQPU4ZA43LWIKPLUUL7HBANCNFSM5ISRULDA&data=04%7C01%7Ckalyn.dorheim%40pnnl.gov%7C1283c24ad9cd4e88417a08d9cf8dd10c%7Cd6faa5f90ae240338c0130048a38deeb%7C0%7C0%7C637769030268474803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=oWEJSZHF9cG%2BG8omkYPsLOCNL93S035Ksf21jM2w9Gk%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Ckalyn.dorheim%40pnnl.gov%7C1283c24ad9cd4e88417a08d9cf8dd10c%7Cd6faa5f90ae240338c0130048a38deeb%7C0%7C0%7C637769030268474803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=BVVqFWIhIB3eWDPu7GMzrL1DK9yPZUWDMeVbO08ANdM%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Ckalyn.dorheim%40pnnl.gov%7C1283c24ad9cd4e88417a08d9cf8dd10c%7Cd6faa5f90ae240338c0130048a38deeb%7C0%7C0%7C637769030268474803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=9t%2FCC2lcVwZuBlEHFSbOpARI6VHFrlHGdHpIF9Li02I%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

kdorheim commented:
@bpbond I believe that I have addressed all of your concerns, are we waiting on @ssmithClimate to take a look?

As of right now none of the tests will work and the manual/vignettes will be outdated. Should I update that stuff as part two of this PR or wait until after this is merged?

kdorheim commented:
@bpbond the baseline ubuntu test we discussed is passing

bpbond commented:
Hurray! Nice work @kdorheim

ssmithClimate commented:
@kdorheim just wanted to check, was the hector input here using RCMIP emission inputs? Thanks!

ssmithClimate commented:
Could you also clarify the difference between the GHG and emissions runs? Is emiss emissions driver for all species (including CO2?), and GHG is emissions driven for air pollutants and concentration driven for all GHGs? Thanks! (Sorry I know you told me this before, but I couldn't remember)

kdorheim commented:
Hi Steve, yes I am using the RCMIP emission inputs here. 

For the GHG runs Hector uses N2O, CH4, and CO2 concentrations instead of emissions as is the case with the other atmospheric forces Hector models.

ssmithClimate commented:
One request, would it be easy to add a figure with total F-gas forcing in the workbook (in addition to the figures for the individual F-gases). It would be useful to compare that total with AR6. Thanks!

kdorheim commented:
Alright, @ssmithClimate I have added a comparison of the aggregate halocarbons please check out https://rpubs.com/kdorheim/853766. Hector underestimates AR6 halocarbon RF by less than 10%, out of the total RF this is a less than 1% difference. 

What do you think about this? Do you think this is a trivial difference?
